tools/runtest

```diff
@@ -776,6 +783,10 @@ else
   done
 fi
 
-exit 0
-
+# shamefully exit if QUICK_CI environment variable is defined and tests are failed.
+if [ ! -z $QUICK_CI ] && [ $ncum_failed_tests -gt 0 ]; then
```

Comment:
Why test for `$QUICK_CI`? Why can't we always exit with 0 if successful and 1 if failed. Any shell script should do that. I often check for $? and use this to decide on what to do next.